params/network_upgrades.go

```diff
@@ -48,22 +56,27 @@ func (n *NetworkUpgrades) forkOrder() []fork {
 // This overrides deactivating the network upgrade by providing a timestamp of nil value.
 func (n *NetworkUpgrades) setDefaults(networkID uint32) {
 	defaults := getDefaultNetworkUpgrades(networkID)
-	if n.SubnetEVMTimestamp == nil {
+	// If the network upgrade is not set, set it to the default value.
+	// If the network upgrade is set to 0, we also treat it as nil and set it default.
+	// This is because in prior versions, upgrades were not modifiable and were directly set to their default values.
+	// Most of the tools and configurations just provide these as 0, so it is safer to treat 0 as nil and set to default
+	// to prevent premature activations of the network upgrades.
```

Comment:
```suggestion
	// to prevent premature activations of the network upgrades for live networks.
```

Reply:
somehow specifying this only applies to testnet / mainnet networks, not local / testing

Reply:
this is because local + testing networks are activated at timestamp 0 already (there is no local or testing definition in https://github.com/ava-labs/avalanchego/blob/master/version/constants.go#L139-L146)